rust_src/src/marker.rs

```diff
@@ -39,6 +39,19 @@ impl LispMarkerRef {
     }
 }
 
+fn build_marker(buf: LispObject, charpos: ptrdiff_t, byte_pos: ptrdiff_t) -> LispObject {
+    let m = Lisp_Marker{
```

Comment:
You'll need to call `allocate_misc` rather than `Lisp_Marker{...}`. As written, this is creating a struct on the stack and returning it, which won't work.